crates/iota-single-node-benchmark/src/mock_storage.rs

```diff
@@ -57,7 +57,7 @@ impl InMemoryObjectStore {
         tx_key: &TransactionKey,
         input_object_kinds: &[InputObjectKind],
     ) -> IotaResult<InputObjects> {
-        let shared_locks_cell: OnceCell<HashMap<_, _>> = OnceCell::new();
+        let shared_locks_cell: OnceCell<Option<HashMap<_, _>>> = OnceCell::new();
```

Comment:
Can you explain this change?

Reply:
Yes. As the `shared_locks` was also changed in the following code.

Reply:
But why?

Reply:
I think it's for better handling when the `shared_locks` returns `None`. When it is `None`, it will be handled by the following error
```rust 
IotaError::GenericAuthority {
    error: "Shared object locks should have been set.".to_string(),
}
```

Reply:
Yes, but it is set by `get_or_init`. This whole change seems very suspicious, as it adds an `Option` only so that it can fail.

Reply:
Tbh I don't have other clues, but I would prefer use this change as we can handle the `None` case better.